### PR TITLE
ceph-deploy doc additions and rework

### PR DESCRIPTION
**Pull request by trhoden**
This make several updates to the docs:

* adds new pages for mds and rgw
* changes the layout of the changelog a bit to make things cleaner/easier to navigate
* fixes some section level mistakes (moving sections up one, down one as far a nesting goes)
* adds release dates in the changelog

All builds and displays correctly for me locally.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -145,7 +145,7 @@
 
 # If true, SmartyPants will be used to convert quotes and dashes to
 # typographically correct entities.
-#html_use_smartypants = True
+html_use_smartypants = False
```
Comment (alfredodeza) [inline docs/source/conf.py]: why this change?

Comment (trhoden) [inline docs/source/conf.py]: Some of the reason is in the commit message: https://github.com/trhoden/ceph-deploy/commit/eec4c3555385d76990b7707d005575c1f309405e

But, we have a lot of section headers that start with double-dashes ``--public-network`` for example.  With smarty pants enabled (default) that gets changed to a unicode long dash.  Looks nice, but not good for code.

To work around it, a lost of place used double-back ticks to make it "pre formatted"  My problem with that was that it looked terrible. See: http://ceph.com/ceph-deploy/docs/pkg.html#install

It's dark on top of dark.  :(

So this was my way to get around that.  I wanted to get rid of the backticks to get rid of the dark on dark, but then it turned the double-dashes into long single dashes.  I don't think disable smartypants has any ill-effect (none that I could see).

File: docs/source/pkg.rst [hunk 2]
```diff
@@ -38,7 +38,7 @@ An example call to install a few packages on 2 hosts (with hostnames like
 
 .. _pkg-remove:
 
-``--remove``
+--remove
```
Comment (alfredodeza) [inline docs/source/pkg.rst]: ah bummer, why remove the back ticks?

Comment (trhoden) [inline docs/source/pkg.rst]: Is it just my browser where the dark gray on top of light gray looks terrible?

http://ceph.com/ceph-deploy/docs/pkg.html#remove

Any place that is a 2nd level header has a light/medium gray background.  The backticks make it then have an additional box around it that is a darker color.  They just don't seem necessary to me at all.

Here is what's there now:

screen shot 2015-03-19 at 5 24 27 pm

On my local build after the changes, this becomes:

screen shot 2015-03-19 at 5 25 05 pm

I think it is better.  But maybe it is personal taste.

Comment (trhoden) [inline docs/source/pkg.rst]: Also, removing the backticks isn't just limited to when there are double-dashes.  I think this looks confusing, for example:

screen shot 2015-03-19 at 5 26 53 pm

I just don't think the backticks add anything useful.

Comment (trhoden) [inline docs/source/pkg.rst]: I could easily pull the backticks and smartypants change (they are intertwined) into a second PR to be debated.  :)
